lib/services/role_test.go

```diff
@@ -1011,8 +1015,187 @@ func TestCheckAccessToServer(t *testing.T) {
 					r.Spec.Allow.Logins = []string{"root"}
 				}),
 			},
-			authPrefMFARequireType: types.RequireMFAType_SESSION,
-			mfaVerified:            true,
+			authSpec: types.AuthPreferenceSpecV2{
+				RequireMFAType: types.RequireMFAType_SESSION,
```

Comment:
Not necessarily within the scope of these changes, but I notice we have no tests here for `RequireMFAType_SESSION_AND_HARDWARE_KEY` or `RequireMFAType_HARDWARE_KEY_TOUCH`

Reply:
cc @Joerger

Reply:
In regards to Per session MFA, `RequireMFAType_SESSION_AND_HARDWARE_KEY`  is functionally equivalent to `RequireMFAType_SESSION` and `RequireMFAType_HARDWARE_KEY_TOUCH`  is functionally equivalent to `RequireMFAType_OFF`. This is unit tested directly in `TestRoleSet_GetAccessState`, instead of adding these equivalent options in every Per session MFA test.

Reply:
Added a couple tests to `TestCheckAccessToServer`, it seems reasonable to me to test it a higher level as well.